lib/node/failure_detector.ml

```diff
@@ -168,3 +170,34 @@ let failure_detection node =
           Lwt_unix.sleep @@ Float.of_int t.config.protocol_period;
         ] in
     Lwt.return ()
+
+module PeerSet = Set.Make (struct
+  let compare = Stdlib.compare
+  type t = Peer.t
+end)
+
+let suspicious_detection node =
```

Comment:
@rosalogia I made a basic function and will need help to make it great 😅 
This function is just run within the `server.run` and simply do, totally async of the failure_detector:
- get all suspicious nodes
- wait for timeout to be reached
- get all suspicious nodes
- intersection of the two lists (to retrieve peers which were suspicious before and after timeout)
- remove the peers present in the intersection

Reply:
Is there a reason the function can't be integrated into the failure detector function?

Reply:
Nope. I just thought it would be better to separate them.

Reply:
Sure :+1: We'll run it right under the current call to failure detector function